lib/mayaHydra/flowViewportAPIExamples/flowViewportAPILocator/mhFlowViewportAPILocator.cpp

```diff
@@ -437,81 +384,80 @@ void FlowViewportAPIMayaLocator::SetupFlowViewportInterfaces()
     }
 }
 
-MStatus FlowViewportAPIMayaLocator::compute( const MPlug& plug, MDataBlock& dataBlock)
+MStatus MhFlowViewportAPILocator::compute( const MPlug& plug, MDataBlock& dataBlock)
 {
-    //Do it only once per call to this function per node
-    
-    if( ! _init){
-        SetCubeGridParametersFromAttributes();
-
-        MObject currentMObj = thisMObject();
-        //Add the callback when an attribute of this node changes
-        _cbAttributeChangedId = MNodeMessage::addAttributeChangedCallback(currentMObj, attributeChangedCallback, ((void*)this));
-
-        //Also monitor parent DAG node to be able to update the scene index if the parent transform is modified
-        MStatus stat;
-        MDagPath parentDagPath = MDagPath::getAPathTo(currentMObj, &stat);
-        CHECK_MSTATUS(stat);
-        parentDagPath.pop();
-        MObject parentObj = parentDagPath.node();
-        _cbParentAttributeChangedId = MNodeMessage::addAttributeChangedCallback(parentObj, transformAttributeChangedCallback, this);
-        
-        _init                               = true;
+    //The MObject can change if the node gets deleted and deletion being undone
+    if (! _thisMObject.isValid()){
+        setupFlowViewportInterfaces();
```

Comment:
Perhaps not in this pull request, but can we move this to the postConstructor() at some point?

Reply:
Yes I am doing it in this PR : https://github.com/Autodesk/maya-hydra/pull/85